specs/customizability.md

```diff
@@ -0,0 +1,350 @@
+
+```rust
+use num_derive::FromPrimitive;
+
+pub const CUSTOMSYSCALL_ACTOR_NAME: &str = "customsyscall";
+
+#[derive(FromPrimitive)]
+#[repr(u64)]
+pub enum Method {
+    Invoke = frc42_dispatch::method_hash!("Invoke"),
+}
+```
+
+[fendermint/actors/customsyscall/src/shared.rs](https://github.com/consensus-shipyard/ipc/blob/98497363a10e08236325e6d5c52755b9fcd52958/fendermint/actors/customsyscall/src/shared.rs#L1)
+
+We next need to write a `[lib.rs](http://lib.rs/)` file which exports the shared code and only compiles `actor.rs` if we are building the Wasm actor:
+
+```
+#[cfg(feature = "fil-actor")]
+mod actor;
+mod shared;
+
+pub use shared::*;
+```
+
+[fendermint/actors/customsyscall/src/lib.rs](https://github.com/consensus-shipyard/ipc/blob/98497363a10e08236325e6d5c52755b9fcd52958/fendermint/actors/customsyscall/src/lib.rs#L1)
+
+<aside>
+💡 NOTE: There are several other files you need to change to compile this actor and package it with the other actors that IPC uses. Please refer to the full example [here](https://github.com/consensus-shipyard/ipc/pull/630) for the following other files you need to change:
+
+- `fendermint/actors/customsyscall/Cargo.toml`: The package for your new actor and all its dependencies
+- `fendermint/actors/Cargo.toml`: Add your new actor as a Wasm target
+- `fendermint/actors/build.rs`: Include your new actor in the `ACTORS` array so it will get included in the bundle.
+- `fendermint/actors/src/manifest.rs`: Add your new actor in the `REQUIRED_ACTORS` array so we can confirm it was correctly bundled on IPC startup
+- `fendermint/vm/actor_interface/src/customsyscall.rs`: A macro which assigns an ID to your new actor and declares constants for accessing it by ID and Address
+- `fendermint/vm/actor_interface/src/lib.rs`: export the constants to IPC
+</aside>
+
+### **7. Load and deploy actor at genesis**
+
+We have so far created a new kernel and syscall, switched IPC to use that kernel and created an actor which calls the new syscall. However, in order to call this actor in IPC, we must load it from the custom_actors_bundle.
+
+To do this open `fendermint/vm/interpreter/src/fvm/genesis.rs` file and in the `init` function add our customsyscall actor right after creating the `chainmetadata` actor:
+
+```rust
+// Initialize the customsyscall actor which gives an example of calling a custom syscall
+state
+    .create_custom_actor(
+        fendermint_actor_customsyscall::CUSTOMSYSCALL_ACTOR_NAME,
+        customsyscall::CUSTOMSYSCALL_ACTOR_ID,
+        &EMPTY_ARR,
+        TokenAmount::zero(),
+        None,
+    )
+    .context("failed to create customsyscall actor")?;
+```
+
+[fendermint/vm/interpreter/src/fvm/genesis.rs#L251](https://github.com/consensus-shipyard/ipc/blob/98497363a10e08236325e6d5c52755b9fcd52958/fendermint/vm/interpreter/src/fvm/genesis.rs#L251)
+
+Your actor has now been deployed and we should be able to send it messages!
+
+# Genesis Parameters
```

Comment:
Should this maybe be a separate spec doc?

Reply:
Yeah, definitely belongs in a separate doc.